### ISSUE TITLE
fails to parse CPP testing MIN_VERSION...

### ISSUE THREAD

**Issue by simonmichael** [rubik/argon#15]
#if !MIN_VERSION_base(4,8,0)

gives: 

    error: 67:22  error: token is not a valid binary operator in a preprocessor subexpression

([context](https://github.com/simonmichael/hledger/blob/master/hledger/Hledger/Cli/CliOptions.hs#L67))

**Comment by rubik** [comment]
Yeah, this is a known issue. Those marcros are provided by Cabal, so I have yet to find a way to make it work. We were discussing some solutions here: #14

**Comment by rubik** [comment]
This is the function we need:
http://haddock.stackage.org/lts-3.10/Cabal-1.22.4.0/Distribution-Simple-BuildPaths.html#v:autogenModulesDir

It gives the path to the autogenerated modules directory, which contains, among other things, the `cabal_macros.h` file that we need. The problem now is how to get the `LocalBuildInfo` object.

**Comment by rubik** [comment]
I confirmed that this bug is fixed with when using the new option `--cabal-macros`. If you are using Stack, it's as easy as:
```
$ argon --cabal-macros $(stack path --dist-dir)/build/autogen/cabal_macros.h src
```

The change has already been pushed to master, and will be included in the upcoming 0.4.0.0 version.
